### PR TITLE
[feat][io] Provide a HTTP Sink

### PR DESCRIPTION
**Pull request by cbornet**
### Motivation

The HTTP Sink pushes records to any HTTP server with the record value in the body of POST method. 
It means you can have event push notification to a simple webhook. This enables the number of applications and platforms to integrate with Pulsar in an existing architecture without Pulsar’s library dependencies and constraints of OS and language that current Pulsar clients can only support.

The Sink takes inspiration from [Pulsar Beam](https://github.com/kafkaesque-io/pulsar-beam) and the [Confluent HTTP Sink connector](https://docs.confluent.io/kafka-connectors/http/current/overview.html).

 The body of the HTTP request is the JSON representation of the record value.
 Some headers are added to the HTTP request:
* `PulsarTopic`: the topic of the record
* `PulsarKey`: the key of the record
* `PulsarEventTime`: the event time of the record
* `PulsarPublishTime`: the publish time of the record
* `PulsarMessageId`: the ID of the message contained in the record
* `PulsarProperties-*`: each record property is passed with the property name prefixed by `PulsarProperties-`

### Modifications

trivial

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:

* Run `HttpSinkTest ` tests

### Does this pull request potentially affect one of the following parts:

no

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] Anything that affects deployment

### Documentation

- [x] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### Matching PR in forked repository

PR in forked repository: https://github.com/cbornet/pulsar/pull/5

### CODE REVIEW

File: pulsar-io/http/pom.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            <groupId>org.apache.pulsar</groupId>
+            <artifactId>pulsar-client-original</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>com.github.tomakehurst</groupId>
+            <artifactId>wiremock-jre8</artifactId>
+            <version>2.33.2</version>
```
Comment (Shoothzj) [inline pulsar-io/http/pom.xml:73]: can we put this version in dependency parent?

**Review by Shoothzj — APPROVED** [review]
better to add new line on the end of **.java** file

File: pulsar-io/http/pom.xml (hunk 1 above)
Comment (cbornet) [reply pulsar-io/http/pom.xml]: done

**Comment by cbornet** [comment]
> better to add new line on the end of .java file

done. PTAL.
Maybe this could be enforced on checkstyle ?

**Review by codelipenghui — CHANGES_REQUESTED** [review]
Hi @cbornet thanks for the contribution
We should start with a proposal since it's a new connector will introduced to Pulsar repo.
And is it possible to move the pulsar-adapters?

**Comment by cbornet** [comment]
Sure, I'll write a PIP.

> And is it possible to move the pulsar-adapters?

I'm not sure pulsar-adapters is the good place if we don't move the other connectors.

**Comment by cbornet** [comment]
Implements [PIP-208: HTTP Sink](https://github.com/apache/pulsar/issues/17719)

**Review by eolivelli — APPROVED** [review]
Great!

LGTM

**Review by codelipenghui — APPROVED** [review]

**Comment by momo-jun** [comment]
@cbornet It will help users a lot if you can add the docs for this feature. Do you have any planned updates on that?

**Comment by cbornet** [comment]
@momo-jun see https://github.com/apache/pulsar/pull/18582 for the doc

**Pull request merged**